FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc

```diff
@@ -267,8 +267,8 @@ std::unique_ptr<fastsim::Particle> fastsim::ParticleManager::nextGenParticle()
             continue;
         }
 
-        // SM particles that descend from exotics and cross the beam pipe radius should make hits but not be decayed 
-        if (producedWithinBeamPipe && !decayedWithinBeamPipe){
+        // SM particles that descend from exotics and cross the beam pipe radius should make hits but not be decayed, by default it will duplicate FastSim hits for long lived particles and so anything produced without activating fixLongLivedBug_ is physically wrong
+        if (producedWithinBeamPipe && !decayedWithinBeamPipe && fixLongLivedBug_){
```

Comment:
I would put the steering `fixLongLivedBug_` first in the list

Reply:
@perrotta , only a small part of particles decay outside beam pipe, so to me logically current variant is OK. From performance point of view, your proposal is faster but likely the difference is negligible, because the expression under if(..) usually is "false".